services/ipfs.go

```diff
@@ -0,0 +1,74 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/ceramicnetwork/go-cas/common/ipfs"
+	"github.com/ceramicnetwork/go-cas/models"
+
+	"github.com/go-playground/validator"
+	iface "github.com/ipfs/boxo/coreiface"
+)
+
+type IpfsService struct {
+	metricService models.MetricService
+	logger        models.Logger
+	ipfsInstances []iface.CoreAPI
```

Comment:
Can you adjust the encapsulation so that the `iface.CoreAPI` is fully contained in `IpfsApi`?

`IpfsApi` is our wrapper around all things IPFS, and `PubSubApi` is an internal wrapper around PubSub-specific things.